src/data.py

```diff
@@ -86,6 +95,46 @@ def empty(self) -> bool:
         """Whether or not the class is empty."""
         return not self._items
 
+    @GObject.Property(type=bool, default=True)
+    def unsaved(self) -> bool:
+        """Whether or not the class is empty."""
+        return self._unsaved
+
+    @unsaved.setter
+    def unsaved(self, unsaved):
+        self._unsaved = unsaved
+        title = \
+            self.get_application().get_window().get_content_title().get_title()
+        if title.startswith("• "):
+            title = title.split("• ")[-1]
+
+        if unsaved:
+            prefix = "• "
+            self.get_application().get_window().get_content_title().set_title(
+                prefix + title)
+            if self.props.project_uri == "":
+                self.get_application().get_window().get_content_title(
+                ).set_subtitle("Draft")
+            else:
+                uri_parse = urlparse(self.props.project_uri)
+                filepath = os.path.abspath(
+                    os.path.join(uri_parse.netloc, uri_parse.path))
+                filepath = filepath.replace(os.path.expanduser("~"), "~")
+                self.get_application().get_window().get_content_title(
+                ).set_subtitle(filepath)
+        else:
+            self.get_application().get_window().get_content_title().set_title(
+                title)
+
+    @GObject.Property(type=str, default="")
+    def project_uri(self) -> str:
+        """Whether or not the class is empty."""
+        return self._project_uri
+
+    @project_uri.setter
+    def project_uri(self, project_uri):
+        self._project_uri = project_uri
```

Comment:
This should be replaced using a simple definition:
`project_uri = GObject.Property(type=str, default="")`